_field-types/flat-object.md

```diff
@@ -126,10 +126,10 @@ PUT /test-index/_doc/2
 ```
 {% include copy-curl.html %}
 
-To search for a leaf value of the flat object, use a POST request. Even if you don't know the field names, you can search for a leaf value in the entire flat object. For example, the following request searches for all issues labeled as bugs:
+To search for a leaf value of the flat object, use either a GET or a POST request. Even if you don't know the field names, you can search for a leaf value in the entire flat object. For example, the following request searches for all issues labeled as bugs:
 
 ```json
-POST /test-index/_search
+GET /test-index/_search
```

Comment:
Why changing the request command from POST to GET? the HTTP POST command is used for queries that have a request body, such as search queries. 

The request body can be large:  search queries can be complex. When using the HTTP GET command, the URL has a limited length, and if the request body is too large, it may not fit in the URL. By using the HTTP POST command, the request body can be sent separately in the request payload, and there is no limit to its size.

Reply:
@mingshl Changed as per @macohen.

Reply:
@mingshl GET is the standard for searching in OpenSearch. it was surprising to me, too, that you can send a payload like you do with POST in a GET. Take a look at https://opensearch.org/docs/latest/aggregations/metric-agg/ and other query docs. the HTTP spec says we're putting meaning where it shouldn't be, but it doesn't disallow it: "A payload within a GET request message has no defined semantics;  sending a payload body on a GET request might cause some existing implementations to reject the request." https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.1